File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/view/LiveIdentityGenerator.java
package com.trtc.uikit.livekit.view;

import androidx.annotation.NonNull;

public final class LiveIdentityGenerator {
    private static final LiveIdentityGenerator INSTANCE = new LiveIdentityGenerator();

    public static LiveIdentityGenerator getInstance() {
        return INSTANCE;
    }

    private LiveIdentityGenerator() {
    }

    public String generateId(String id, @NonNull RoomType type) {
        return type.getPrefix() + id;
    }

    public RoomType getIDType(String id) {
        if (id == null) {
            return null;
        }
        for (RoomType type : RoomType.values()) {
            if (id.startsWith(type.getPrefix())) {
                return type;
            }
        }
        return null;
    }

    public enum RoomType {
        LIVE("live_"),
        VOICE("voice_");

        private final String prefix;

        RoomType(String prefix) {
            this.prefix = prefix;
        }

        public String getPrefix() {
            return prefix;
        }
    }
}

File: Flutter/lib/widget/live_identity_generator.dart
class LiveIdentityGenerator {
  static final LiveIdentityGenerator instance = LiveIdentityGenerator._internal();

  factory LiveIdentityGenerator() {
    return instance;
  }

  LiveIdentityGenerator._internal();

  String generateId(String id, RoomType type) {
    return type.prefix + id;
  }

  RoomType? getIDType(String? id) {
    if (id == null) {
      return null;
    }
    for (RoomType type in RoomType.values) {
      if (id.startsWith(type.prefix)) {
        return type;
      }
    }
    return null;
  }
}

enum RoomType {
  live('live_'),
  voice('voice_');

  final String prefix;

  const RoomType(this.prefix);
}
